### PR TITLE
feat(aws): Added AWS role session name parameter

### PR DESCRIPTION
**Pull request by Fennerr**
### Context

Added as per this request in the prowler slack: https://prowler-workspace.slack.com/archives/C0451NDLC4X/p1704113278185009

### Description

Added a parameter to specify the RoleSessionName value passed to AWS api when performing assume-role calls.

I have not tested this myself. I have requested that the requestee tests it out.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/lib/arguments/arguments.py [hunk 1]
```diff
@@ -27,6 +27,13 @@ def init_parser(self):
         help="ARN of the role to be assumed",
         # Pending ARN validation
     )
+    aws_auth_subparser.add_argument(
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/arguments/arguments.py:30]: We have to include tests for this new argument.

File: prowler/providers/aws/lib/arguments/arguments.py [hunk 2]
```diff
@@ -203,3 +210,16 @@ def validate_bucket(bucket_name):
         raise ArgumentTypeError(
             "Bucket name must be valid (https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)"
         )
+
+
+def validate_role_session_name(session_name):
```
Comment (jfagoagas) [inline prowler/providers/aws/lib/arguments/arguments.py:215]: We have to include tests for this new function.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Thanks for this new flag @Fennerr ! I left some comments to include tests for the new argument and function to validate it. We'll do that as soon as we can.

**Comment by Fennerr** [comment]
Okay awesome. Something I just realised is that I didnt cover setting a role name for the call to assume role when using an Organizations Role.

For now I commited a change to assume_role:
```python
role_session_name = (
    assumed_role_info.role_session_name
    if assumed_role_info.role_session_name
    else "ProwlerAsessmentSession"
)
``` 

Just to ensure that any calls to assume_role will be guaranteed to have a role session name

This is why tests are important!

**Review by jfagoagas — APPROVED** [review]
Awesome!! 

Thanks @Fennerr and @sergargar !! 🚀

**Comment by MrCloudSec** [comment]
Testing and docs were added!

**Pull request merged**